src/specsynthase/specbuilder.py

```diff
@@ -49,3 +50,7 @@ def add_spec(self, file_name):
 
     def dump(self, *args, **kwargs):
         return yaml.safe_dump(dict(**self), *args, **kwargs)
+
+    def validate(self):
+        validate_spec(self)
```

Comment:
I see you've already filed #8. This brings up a good question though: how should support for different versions of the spec look like? If there will be separate classes, this is probably a fine way to do validation. If multiple versions of the spec need to be supported by this class, you may need to do something different here. That's not a question that needs answering before this can be merged, just something to keep in mind.